src/formatDate.js

```diff
@@ -8,7 +8,29 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[3]);
+  const newDate = [];
+  const dateSet = {};
+
+  for (let i = 0; i < dateParts.length; i++) {
+    dateSet[fromFormat[i]] = dateParts[i];
+  }
+
+  for (const key of toFormat) {
+    if (dateSet[key]) {
```

Comment:
The check `if (dateSet[key])` may not work as expected if the date part is `00` (e.g., the day or month). In JavaScript, `0` is falsy, and if any part of the date is `00`, it will be skipped. You should explicitly check if the key exists in the `dateSet` object.